src/components/Video/VideoContent/VideoContent.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import videojs from 'video.js';
+import Component from 'video.js/dist/types/component.d';
+import Player from 'video.js/dist/types/player.d';
+import classNames from 'classnames';
+import { useSearchParams } from 'next/navigation';
+import styles from './VideoContent.module.scss';
+import LazyLoad from '../../LazyLoad/LazyLoad';
+import useSharedUnmutedVideoState from '../useSharedUnmutedVideoState';
+import 'video.js/dist/video-js.css';
+import Controls from '../Controls/Controls';
+import { VideoData } from '../../../types/types';
+
+export type VideoContentProps = {
+  hasControls: boolean;
+  isAutoplaying: boolean;
+  isLooping: boolean;
+  video: VideoData;
+};
+
+type VideoJsComponent = Component & {
+  handleClick: () => void;
+};
+
+const VideoContent = ({
+  isAutoplaying,
+  hasControls,
+  isLooping,
+  video: { baseUrl, blur, height, hls, width },
+}: VideoContentProps) => {
+  const videoContainerRef = useRef(null);
+
+  const searchParams = useSearchParams();
+  const hasFocusMode = searchParams.get('hasFocusMode');
+
+  const [player, setPlayer] = useState<Player>(null);
+  const [isPlaying, setIsPlaying] = useState(isAutoplaying);
+  const [hasLoaded, setHasLoaded] = useState(false);
+
+  const [isMuted, toggleIsMuted] = useSharedUnmutedVideoState(hls ?? 'unknown');
+
+  const handlePlayToggle = () => {
```

Comment:
Bit confused why we have this callback but also a useEffect at line 100 which does something very similar but through the isPlaying state. Feels a bit like mixing uncontrolled and controlled input fields in a React form, but maybe it's correct and just needs some explanation...

Reply:
I just moved the Video component to VideoContent, should have flagged that in advance. Agree though, this needs tidying up.

I will make a separate ticket to pick up your comments in there.